### ISSUE TITLE
Datadog: Include request times in dashboard

### ISSUE THREAD

**Issue by lukey-luke** [department-of-veterans-affairs/abd-vro#1724]
Timeboxing this ticket. If complications arise, we can create follow up tickets.

**User Story**

As an engineer, I want view request time for contention classification, so that I understand performance, which impacts how long veterans are sitting there waiting.

**Acceptance Criteria**

1. Average request time is made available in Contention Classification view of Datadog dashboard [HERE](https://lighthousedi.ddog-gov.com/dashboard/pvw-4a2-2s5/vro-on-lhdi?tpl_var_cluster%5B0%5D=%2A&tpl_var_deployment%5B0%5D=vro-domain-cc&tpl_var_namespace%5B0%5D=%2A&from_ts=1687279193884&to_ts=1687282793884&live=true).
2. Max request time is made available in Contention Classification view of Datadog dashboard [HERE](https://lighthousedi.ddog-gov.com/dashboard/pvw-4a2-2s5/vro-on-lhdi?tpl_var_cluster%5B0%5D=%2A&tpl_var_deployment%5B0%5D=vro-domain-cc&tpl_var_namespace%5B0%5D=%2A&from_ts=1687279193884&to_ts=1687282793884&live=true).
3. If this info is already available, message @lukey-luke and tell him how to RTFM.

**Not included in this work**

A note about things that should not be considered necessary.

**Notes about work**
- _(links to documentation, resources, code, etc that might inform the person doing the work)_

**Comment by yurenanava** [comment]
Bring to the next refinement session with VRO

**Comment by lukey-luke** [comment]
Marking blocked, need LHDI to support VRO + us to access custom metrics
FYSA @tejans24

**Comment by verdance-luke** [comment]
blocked because need write access for LHDI datadog

**Comment by dianagriffin** [comment]
Following up on the two above blockers: 
- LHDI support for custom metrics unblocked here: https://dsva.slack.com/archives/C03UA9MV1EH/p1698864951430509 
- Instructions to request write access for LHDI datadog shared here: https://dsva.slack.com/archives/C01Q7979Z7D/p1700170933416979

**Comment by lukey-luke** [comment]
marked unblocked, will have to get LHDI access first, associated ticket:
https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/313

**Comment by dianagriffin** [comment]
@verdance-luke cc/ @yurenanava @msnwatson I've unassigned this from VRO because I believe when we last discussed it we aligned that it makes sense for CC to go ahead with setting up a CC-specific implementation to get your immediate needs met. Feel free to bounce it back to me if that's incorrect!

**Comment by verdance-luke** [comment]
might be able to get this from APM in datadog... posted q in vro-support on accessing config file [HERE](https://dsva.slack.com/archives/C01Q7979Z7D/p1703028481434769)

**Comment by verdance-luke** [comment]
Reached out to @tejans24 on VRO to get up-to-date

Looks like we can use [datadogpy](https://github.com/DataDog/datadogpy) to access dogstatsd as well as the datadog API. This could be used to track a new custom metric for datadog. We could integrate this to our FastAPI app as a [middleware](https://fastapi.tiangolo.com/tutorial/middleware/). That would be a good solution to track across all request in python.

However, I'd like to better understand the API gateway that routes traffic to services like the contention classifier API and see if we can do this even one level higher, making it a more general purpose solution

**Comment by verdance-luke** [comment]
Submitted support request ticket to VRO [here](https://github.com/department-of-veterans-affairs/abd-vro/issues/2387).

**Comment by yurenanava** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Contention-Classification-640a24db5dcd08457cebbc79/issues/department-of-veterans-affairs/abd-vro/1724?planning-poker) with Zenhub @tyler-spangler6 @verdance-luke

**Comment by verdance-luke** [comment]
Need to review dogstatsd documentation Mason linked in the VRO support slack:
https://dsva.slack.com/archives/C01Q7979Z7D/p1705094314458359?thread_ts=1705091879.039769&cid=C01Q7979Z7D

**Comment by verdance-luke** [comment]
Pushed some code up to get started in [feature-1724-log-request-times](https://github.com/department-of-veterans-affairs/abd-vro/tree/feature-1724-log-request-times).

Will follow up w/ VRO on slack to figure out 1.) how to connect to DEV stats container for testing or 2.) how to coordinate a deployment to a lower environment for testing.

**Comment by verdance-luke** [comment]
I think the plan is to deploy a new branch to DEV environment in VRO

**Comment by verdance-luke** [comment]
Looks like deploy [process for a custom branch to dev](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Deploying-VRO#deploy-to-dev-using-a-custom-branch) is a little bit different. I posted a thread in the vro support slack [here](https://dsva.slack.com/archives/C01Q7979Z7D/p1707429386298359).

**Comment by raudabaugh** [comment]
Heads up, LHDI is requiring teams to use the REST API instead of statsd:
https://dsva.slack.com/archives/C03UA9MV1EH/p1707342096814809

**Comment by verdance-luke** [comment]
Looks like following [this thread](https://dsva.slack.com/archives/C01Q7979Z7D/p1707432385994699) on slack, EE team was able to use REST API.

I did add statsd call and logging statement to `@app.middleware` in api.py, but not seeing new logging statements for calls to dev.lighthouse.va.gov/. I will try to re-deploy ` feature-1724-log-request-times` w/ additional logging to ensure that remote is pushed up

**Comment by verdance-luke** [comment]
Submitted request to get Datadog api key (REQ22113940 - - - New Monitoring) per LHDI docs [here](https://laughing-adventure-wllepqe.pages.github.io/Lighthouse-delivery-infrastructure-tools/Using-the-DI-tools/Step-6-Observability/Datadog-Observability-Tools/).

**Comment by verdance-luke** [comment]
follow-up issue [here](https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/379)
